### PR TITLE
Comp Req inspection - Bitmanipulation

### PR DESCRIPTION
**Pull request by aschemmel-tech**
Resolves: #2479

### CODE REVIEW

File: docs/modules/baselibs/bitmanipulation/docs/requirements/index.rst [hunk 1]
```diff
@@ -36,6 +36,7 @@ Functional Requirements
    :safety: ASIL_B
    :satisfies: feat_req__baselibs__bitmanipulation, feat_req__baselibs__core_utilities
    :status: valid
+   :tags: inspected
```
Comment (PandaeDo) [inline docs/modules/baselibs/bitmanipulation/docs/requirements/index.rst:40]: Why there is a tag inspected? It seems to me that the inspection isn't done yet.

Comment (aschemmel-tech) [reply docs/modules/baselibs/bitmanipulation/docs/requirements/index.rst:40]: But it will be after the inspection when the checklist is filled out. This is only to document the scope. But you are too early in reviewing, I will set the status back to draft, as the content of the PR is not complete.

File: docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+      * - REQ_07_02
+        - Is the attribute *security* set correctly?
+        - For component requirements this checklist item is supported by automated check: "Every requirement which satisfies a feature requirement with security attribute set to YES inherits this". But the component requirements/architecture may additionally also be subject to a :need:`wp__sw_component_security_analysis`.
+        -
+        -
+        -
+      * - REQ_08_01
+        - Is the requirement *verifiable*?
+        - If at the time of the inspection already tests are created for the requirement, the answer is yes. This can be checked via traces, but also :need:`gd_req__req_attr_test_covered` shows this. In case the requirement is not sufficiently traced to test cases already, a test expert is invited to the inspection to give their opinion whether the requirement is formulated in a way that supports test development and the available test infrastructure is sufficient to perform the test.
+        -
```
Comment (rahulthakre29) [inline docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst]: YES

File: docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+        -
+      * - REQ_07_02
+        - Is the attribute *security* set correctly?
+        - For component requirements this checklist item is supported by automated check: "Every requirement which satisfies a feature requirement with security attribute set to YES inherits this". But the component requirements/architecture may additionally also be subject to a :need:`wp__sw_component_security_analysis`.
+        -
+        -
+        -
+      * - REQ_08_01
+        - Is the requirement *verifiable*?
```
Comment (rahulthakre29) [inline docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst:151]: Yes, all requirements have test cases implemented

Comment (aschemmel-tech) [reply docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst:151]: included your comment in checklist

File: docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+
+      * - Review ID
+        - Acceptance Criteria
+        - Guidance
+        - Passed
+        - Remarks
+        - Issue link
+      * - REQ_01_01
+        - Is the requirement formulation template used?
+        - see :need:`gd_temp__req_formulation`, this includes the use of "shall".
```
Comment (mihajlo-k) [inline docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst:56]: Passed: **NO** (2/9 failed)

Following requirements failed to comply with the requirements formulation template:

- [aou_req__bitmanipulation__type_constraints](https://github.com/eclipse-score/score/blob/main/docs/modules/baselibs/bitmanipulation/docs/requirements/index.rst?plain=1#L98): The second sentence does not use `shall` and is not marked with `Note`. Suggestion: move it to the `Note`.
- [aou_req__bitmanipulation__concurrent_access](https://github.com/eclipse-score/score/blob/main/docs/modules/baselibs/bitmanipulation/docs/requirements/index.rst?plain=1#L116): **"…as the library provides no internal thread safety guarantees."** is a justification and not a verifiable statement. Per the template, it should be separated and marked as `Note`.

The remaining requirements use the template correctly.

File: docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+        -
+        -
+        -
+      * - REQ_02_01
+        - Is the requirement description *comprehensible* ?
+        - If you think the requirement is hard to understand, comment here.
```
Comment (mihajlo-k) [inline docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst:65]: Passed: **NO** (1/9 failed)

Following requirement is not clear considering it's an AoU:

- [aou_req__bitmanipulation__valid_bit_positions](https://github.com/eclipse-score/score/blob/main/docs/modules/baselibs/bitmanipulation/docs/requirements/index.rst?plain=1#L89): This requirement seems like a functional one. It basically describes what the _library_ guarantees, and might be a duplicate of [comp_req__bitmanipulation__bounds_safety](https://github.com/eclipse-score/score/blob/main/docs/modules/baselibs/bitmanipulation/docs/requirements/index.rst?plain=1#L63). It should be clarified what is meant by this AoU requirement.

The remaining requirements are clear and easy to understand. However, some grammar mistakes are still present:

- [comp_req__bitmanipulation__utilities](https://github.com/eclipse-score/score/blob/main/docs/modules/baselibs/bitmanipulation/docs/requirements/index.rst?plain=1#L41): "...shall provide API..." -> "...shall provide **an** API..."
- [aou_req__bitmanipulation__type_constraints](https://github.com/eclipse-score/score/blob/main/docs/modules/baselibs/bitmanipulation/docs/requirements/index.rst?plain=1#L98): "...shall **only** use bit manipulation functions with integral types..." -> "...shall use bit manipulation functions **only** with integral types..."
- [aou_req__bitmanipulation__enum_type_safety](https://github.com/eclipse-score/score/blob/main/docs/modules/baselibs/bitmanipulation/docs/requirements/index.rst?plain=1#L107): Missing period at the end

File: docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+        -
+        -
+        -
+      * - REQ_02_02
+        - Is the requirement description *unambiguous* ?
+        - Especially search for "weak words" like "about", "etc.", "relevant" and others (see the internet documentation on this). This check shall be supported by tooling.
```
Comment (mihajlo-k) [inline docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst:75]: Passed: **NO** (3/9 failed)

Following requirements have some ambiguities/weak words:

- [comp_req__bitmanipulation__utilities](https://github.com/eclipse-score/score/blob/main/docs/modules/baselibs/bitmanipulation/docs/requirements/index.rst?plain=1#L41): "**...manipulating half-bytes and bytes...**" what is meant by **manipulating**? E.g. different implementers could have different ideas what should be done. This looks especially ambiguous related to the enumeration of all the bit-related operations in the first part of the requirement ("...setting, clearing, toggling, and checking bits...").
- [comp_req__bitmanipulation__bounds_safety](https://github.com/eclipse-score/score/blob/main/docs/modules/baselibs/bitmanipulation/docs/requirements/index.rst?plain=1#L63): **"...prevent data corruption"** is too vague. Which data? What is meant by corruption?
- [aou_req__bitmanipulation__enum_type_safety](https://github.com/eclipse-score/score/blob/main/docs/modules/baselibs/bitmanipulation/docs/requirements/index.rst?plain=1#L107): Usage of a "weak word" explicitly mentioned in the checklist: **etc.**

The remaining requirements are clear and unambiguous.

File: docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+        -
+        -
+        -
+      * - REQ_02_03
+        - Is the requirement description *atomic* ?
+        - A good way to think about this is to consider if the requirement may be tested by one (positive) test case or needs more of these. The requirement formulation template should also avoid being non-atomic already. Note that there are cases where also non-atomic requirements are the better ones, for example if those are better understandable.
```
Comment (mihajlo-k) [inline docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst:85]: Passed: **NO** (1/9 failed)

Following requirement is not atomic:

- [comp_req__bitmanipulation__utilities](https://github.com/eclipse-score/score/blob/main/docs/modules/baselibs/bitmanipulation/docs/requirements/index.rst?plain=1#L41): This requirement bundles at least four distinct functional areas: (1) setting/clearing/toggling/checking bits, (2) extracting bytes, (3) manipulating half-bytes, (4) manipulating bytes. Each area requires multiple independent test cases. For example, "set bit" and "extract byte" are entirely unrelated operations with different signatures, parameters, and edge cases. A single positive test case cannot cover all of them. Consider splitting into separate requirements per operation group.

The remaining requirements are reasonably atomic.

File: docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst [hunk 8]
```diff
@@ -0,0 +1,172 @@
...
+        -
+        -
+        -
+      * - REQ_02_04
+        - Is the requirement description *feasible* ?
+        - If at the time of the inspection the requirement has already some implementation, the answer is yes. This can be checked via traces, but also :need:`gd_req__req_attr_impl` shows this. In case the requirement has no implementation at the time of inspection (i.e. not implemented at least as "proof-of-concept"), a development expert should be invited to the Pull-Request review to explicitly check this item.
```
Comment (mihajlo-k) [inline docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst:94]: Passed: **YES**

All requirements are already implemented at the time of this inspection

File: docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst [hunk 9]
```diff
@@ -0,0 +1,172 @@
...
+        -
+        -
+        -
+      * - REQ_02_05
+        - Is the requirement description *independent from implementation* ?
+        - This checkpoint should improve requirements definition in the sense that the "what" is described and not the "how" - the latter should be described in architecture/design derived from the requirement. But there can also be a good reason for this, for example we would require using a file format like JSON and even specify the formatting standard already on stakeholder requirement level because we want to be compatible. A finding in this checkpoint does not mean there is a safety problem in the requirement.
```
Comment (mihajlo-k) [inline docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst:100]: Passed: **YES**

Note on [comp_req__bitmanipulation__header_only](https://github.com/eclipse-score/score/blob/main/docs/modules/baselibs/bitmanipulation/docs/requirements/index.rst?plain=1#L77): It's describing an implementation constraint, so it is a "how". However, this is a justified architectural decision, and it's analogous to the "JSON format" example from the checklist.

The remaining requirements are independent from implementation.

File: docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst [hunk 10]
```diff
@@ -0,0 +1,172 @@
...
+        -
+        -
+        -
+      * - REQ_03_01
+        - Is the *linkage to the parent feature/component requirement* correct?
+        - Linkage to correct levels and ASIL attributes is checked automatically, but it needs checking if the child requirement implements (at least) a part of the parent requirement.
```
Comment (mihajlo-k) [inline docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst:106]: Passed: **YES**

The linkage to parent requirements is correct for all the requirements.

File: docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst [hunk 11]
```diff
@@ -0,0 +1,172 @@
...
+        -
+        -
+        -
+      * - REQ_04_01
+        - Is the requirement *internally and externally consistent*?
+        - Does the requirement contradict other requirements within the same or higher levels? One may restrict the search to the feature for component requirements, for features to other features using same components.
```
Comment (mihajlo-k) [inline docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst:112]: Passed: **YES**

All the requirements are mutually consistent.

However, there are inconsistencies between some requirements' titles and their descriptions:

- [aou_req__bitmanipulation__concurrent_access](https://github.com/eclipse-score/score/blob/main/docs/modules/baselibs/bitmanipulation/docs/requirements/index.rst?plain=1#L109): The title **"No Side Effects on Concurrent Access"** is misleading. It sounds like the library guarantees no side effects during concurrent access (i.e., it is thread-safe). The text says the opposite — the library has no thread safety, and the user must synchronize.
- [aou_req__bitmanipulation__enum_type_safety](https://github.com/eclipse-score/score/blob/main/docs/modules/baselibs/bitmanipulation/docs/requirements/index.rst?plain=1#L100): **"...Type Safety"** is misleading — the requirement is about value constraints, not type safety.

File: docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst [hunk 12]
```diff
@@ -0,0 +1,172 @@
...
+        -
+        -
+        -
+      * - REQ_05_01
+        - Do the software requirements consider *timing constraints*?
+        - This checkpoint encourages to think about timing constraints even if those are not explicitly mentioned in the parent requirement. If the reviewer of a requirement already knows or suspects that the code execution will be consuming a lot of time, one should think of the expectation of a "user".
```
Comment (mihajlo-k) [inline docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst:121]: Passed: **YES**

The overhead for bit manipulation operations is negligible.

File: docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst [hunk 13]
```diff
@@ -0,0 +1,172 @@
...
+        -
+      * - REQ_06_01
+        - Does the requirement consider *external interfaces*?
+        - The SW platform's external interfaces (to the user) are defined in the Feature Architecture, so the Feature and Component Requirements should determine the input data use and setting of output data for these interfaces. Are all output values defined?
+        -
+        -
+        -
+      * - REQ_07_01
+        - Is the *safety* attribute set correctly?
+        - Derived requirements are checked automatically, see :need:`gd_req__req_linkage_safety`. But for the top level requirements (and also all AoU) this needs to be checked manually for correctness.
```
Comment (mihajlo-k) [inline docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst:140]: Passed: **YES**

All safety attributes set correctly.

File: docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst [hunk 14]
```diff
@@ -0,0 +1,172 @@
...
+        -
+        -
+        -
+      * - REQ_07_02
+        - Is the attribute *security* set correctly?
+        - For component requirements this checklist item is supported by automated check: "Every requirement which satisfies a feature requirement with security attribute set to YES inherits this". But the component requirements/architecture may additionally also be subject to a :need:`wp__sw_component_security_analysis`.
```
Comment (mihajlo-k) [inline docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst:146]: Passed: **YES**

All security attributes set correctly.

File: docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst [hunk 15]
```diff
@@ -0,0 +1,172 @@
...
+        - Is the requirement *verifiable*?
+        - If at the time of the inspection already tests are created for the requirement, the answer is yes. This can be checked via traces, but also :need:`gd_req__req_attr_test_covered` shows this. In case the requirement is not sufficiently traced to test cases already, a test expert is invited to the inspection to give their opinion whether the requirement is formulated in a way that supports test development and the available test infrastructure is sufficient to perform the test.
+        - YES
+        - @rahulthakre29: all equirements have test cases implemented
+        -
+      * - REQ_08_02
+        - Is the requirement verifiable by design or code review in case it is not feasibly testable?
+        - In very rare cases a requirement may not be verifiable by test cases, for example a specific non-functional requirement. In this case a requirement analysis verifies the requirement by design/code review. If such a requirement is in scope of this inspection, please check this here and link to the respective review record. A test expert is invited to the inspection to confirm their opinion that the requirement is not testable.
```
Comment (mihajlo-k) [inline docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst:158]: Passed: **YES**

Only one requirement is relevant for this inspection - [comp_req__bitmanipulation__header_only](https://github.com/eclipse-score/score/blob/main/docs/modules/baselibs/bitmanipulation/docs/requirements/index.rst?plain=1#L77) and it's very well verifiable by design/code review.

File: docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst [hunk 16]
```diff
@@ -0,0 +1,172 @@
...
+        -
+        -
+        -
+      * - REQ_09_01
+        - Do the requirements that define a safety mechanism specify the error reaction leading to a safe state?
+        - Alternatively to the safe state there could also be "repair" mechanisms. Also do not forget to consider REQ_05_01 for these.
```
Comment (mihajlo-k) [inline docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst:164]: Passed: **NO** (failed 1/1)

There is only one requirement here that _does_ define a safety mechanism - [comp_req__bitmanipulation__bounds_safety](https://github.com/eclipse-score/score/blob/main/docs/modules/baselibs/bitmanipulation/docs/requirements/index.rst?plain=1#L63), but it fails to specify error reaction leading to a safe state (e.g. abort or return error indicator). Just **"...prevent data corruption"** is not enough.

File: docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst [hunk 17]
```diff
@@ -0,0 +1,172 @@
...
+        -
+      * - REQ_05_01
+        - Do the software requirements consider *timing constraints*?
+        - This checkpoint encourages to think about timing constraints even if those are not explicitly mentioned in the parent requirement. If the reviewer of a requirement already knows or suspects that the code execution will be consuming a lot of time, one should think of the expectation of a "user".
+        -
+        -
+        -
+      * - REQ_06_01
+        - Does the requirement consider *external interfaces*?
+        - The SW platform's external interfaces (to the user) are defined in the Feature Architecture, so the Feature and Component Requirements should determine the input data use and setting of output data for these interfaces. Are all output values defined?
```
Comment (mihajlo-k) [inline docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst:127]: Passed: **NO** (3/9 failed)

For following requirements there is no description of the output:

- [comp_req__bitmanipulation__utilities](https://github.com/eclipse-score/score/blob/main/docs/modules/baselibs/bitmanipulation/docs/requirements/index.rst?plain=1#L41): The requirement describes inputs but not outputs — no return types or return semantics specified.
- [comp_req__bitmanipulation__bounds_safety](https://github.com/eclipse-score/score/blob/main/docs/modules/baselibs/bitmanipulation/docs/requirements/index.rst?plain=1#L63): Describes input validation but no output on failure. When an interface operation receives invalid input, what is returned? The error output path is undefined.

Additionally, following requirements show mismatch with the architecture:

- [comp_req__bitmanipulation__utilities](https://github.com/eclipse-score/score/blob/main/docs/modules/baselibs/bitmanipulation/docs/requirements/index.rst?plain=1#L41): **"extracting bytes and manipulating half-bytes"** have no corresponding interface operations in the architecture.
- [aou_req__bitmanipulation__bit_validation](https://github.com/eclipse-score/score/blob/main/docs/modules/baselibs/bitmanipulation/docs/requirements/index.rst?plain=1#L125): References **"byte and half-byte extraction indices"** which have no corresponding interface operations in the architecture.

The remaining requirements are compliant.

File: docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst (hunk 4 above)
Comment (aschemmel-tech) [reply docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst:56]: comment taken over in "Remarks"

File: docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst (hunk 5 above)
Comment (aschemmel-tech) [reply docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst:65]: comment taken over in "Remarks"

File: docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst (hunk 6 above)
Comment (aschemmel-tech) [reply docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst:75]: comment taken over in "Remarks"

File: docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst (hunk 7 above)
Comment (aschemmel-tech) [reply docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst:85]: comment taken over in "Remarks"

File: docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst (hunk 8 above)
Comment (aschemmel-tech) [reply docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst:94]: comment taken over in "Remarks"

File: docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst (hunk 9 above)
Comment (aschemmel-tech) [reply docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst:100]: comment taken over in "Remarks"

File: docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst (hunk 10 above)
Comment (aschemmel-tech) [reply docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst:106]: comment taken over in "Remarks"

File: docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst (hunk 11 above)
Comment (aschemmel-tech) [reply docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst:112]: comment taken over in "Remarks"

File: docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst (hunk 12 above)
Comment (aschemmel-tech) [reply docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst:121]: comment taken over in "Remarks"

File: docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst (hunk 13 above)
Comment (aschemmel-tech) [reply docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst:140]: comment taken over in "Remarks"

File: docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst (hunk 14 above)
Comment (aschemmel-tech) [reply docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst:146]: comment taken over in "Remarks"

File: docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst (hunk 15 above)
Comment (aschemmel-tech) [reply docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst:158]: comment taken over in "Remarks"

File: docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst (hunk 16 above)
Comment (aschemmel-tech) [reply docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst:164]: comment taken over in "Remarks"

File: docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst (hunk 17 above)
Comment (aschemmel-tech) [reply docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst:127]: comment taken over in "Remarks"

File: docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst (hunk 4 above)
Comment (mihajlo-k) [reply docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst:56]: Passed: **YES**

all findings addressed in [this PR](https://github.com/eclipse-score/score/pull/2677).

File: docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst (hunk 5 above)
Comment (mihajlo-k) [reply docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst:65]: Passed: **YES**

all findings addressed in [this PR](https://github.com/eclipse-score/score/pull/2677).

File: docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst (hunk 6 above)
Comment (mihajlo-k) [reply docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst:75]: Passed: **YES**

all findings addressed in [this PR](https://github.com/eclipse-score/score/pull/2677).

File: docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst (hunk 7 above)
Comment (mihajlo-k) [reply docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst:85]: Passed: **YES**

all findings addressed in [this PR](https://github.com/eclipse-score/score/pull/2677).

File: docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst (hunk 11 above)
Comment (mihajlo-k) [reply docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst:112]: Even though not directly part of the inspection checklist all the inconsistencies between the requirements' titles and their descriptions are corrected in [this PR](https://github.com/eclipse-score/score/pull/2677).

File: docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst (hunk 17 above)
Comment (mihajlo-k) [reply docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst:127]: Passed: **YES**

all findings addressed in [this PR](https://github.com/eclipse-score/score/pull/2677).

**NOTE:** resolving the issues with missing interfaces in the component architecture is not in the scope of requirements inspection.

File: docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst (hunk 16 above)
Comment (mihajlo-k) [reply docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst:164]: Passed: **YES**

all findings addressed in [this PR](https://github.com/eclipse-score/score/pull/2677).

Comment (aschemmel-tech) [reply docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst:164]: I do not think this finding was properly fixed - see comment in the checklist.

File: docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst (hunk 4 above)
Comment (aschemmel-tech) [reply docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst:56]: modified checklist accordingly

File: docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst (hunk 5 above)
Comment (aschemmel-tech) [reply docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst:65]: modified checklist accordingly

File: docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst (hunk 6 above)
Comment (aschemmel-tech) [reply docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst:75]: modified checklist accordingly

File: docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst (hunk 7 above)
Comment (aschemmel-tech) [reply docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst:85]: modified checklist accordingly

File: docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst (hunk 11 above)
Comment (aschemmel-tech) [reply docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst:112]: modified checklist accordingly

File: docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst (hunk 17 above)
Comment (aschemmel-tech) [reply docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst:127]: modified checklist accordingly

File: docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst (hunk 16 above)
Comment (aschemmel-tech) [reply docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst:164]: fixed in https://github.com/eclipse-score/score/pull/2696

File: docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst [hunk 18]
```diff
@@ -0,0 +1,195 @@
...
+        - Passed
+        - Remarks
+        - Issue link
+      * - REQ_01_01
+        - Is the requirement formulation template used?
+        - see :need:`gd_temp__req_formulation`, this includes the use of "shall".
+        - YES
+        - Following requirements failed to comply with the requirements formulation template:
+
+          - :need:`aou_req__bitmanipulation__type_constraints` - The second sentence does not use shall and is not marked with "Note". Suggestion - move it to the Note.
```
Comment (4og) [inline docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst]: Can you please remove linking to requirements via `:need:` as those are already fixed in PR-2677?
Otherwise, it is somewhat misleading for those who read the checklist page.

Comment (aschemmel-tech) [reply docs/modules/baselibs/bitmanipulation/docs/requirements/chklst_req_inspection.rst]: I will rework the checklist entries to not mention already fixed topics.
